From a2185ac3504b1fdd1fbf1666fd098a11882bfce5 Mon Sep 17 00:00:00 2001
From: Jakub Kicinski <kubakici@wp.pl>
Date: Wed, 11 Dec 2013 18:09:59 +0100
Subject: [PATCH 01/11] Fix diff format detection.

When '-p' option is used ' @@' can be followed by function name.
---
 src/textview.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/textview.c b/src/textview.c
index 7372a1c21..72f0efc90 100644
--- a/src/textview.c
+++ b/src/textview.c
@@ -1630,7 +1630,7 @@ static void textview_write_line(TextView *textview, const gchar *str,
 			else if (buf[0] == '-')
 				fg_color = "diff-del";
 			else if (strncmp(buf, "@@ ", 3) == 0 &&
-					strcmp(buf+strlen(buf)-4, " @@\n") == 0)
+				 strstr(&buf[3], " @@"))
 				fg_color = "diff-hunk";
 		} else if (strcmp(buf,"-- \n") == 0
 				|| strcmp(buf, "- -- \n") == 0
-- 
2.14.1

